latch_cli/services/register/register.py

```diff
@@ -308,33 +324,42 @@ def register(
             click.secho("Skipping confirmation because of --yes", bold=True)
 
         click.secho("Initializing registration", bold=True)
+        transport = None
+        scp = None
+
         if remote:
             print("Connecting to remote server for docker build...")
 
+            assert ctx.ssh_client is not None
+            transport = ctx.ssh_client.get_transport()
+
+            assert transport is not None
+            scp = SCPClient(transport=transport, sanitize=lambda x: x)
+
         with contextlib.ExitStack() as stack:
-            td = stack.enter_context(MaybeRemoteDir(ctx.ssh_client))
+            td: str = stack.enter_context(MaybeRemoteDir(ctx.ssh_client))
             _build_and_serialize(
                 ctx,
                 ctx.default_container.image_name,
                 ctx.default_container.pkg_dir,
                 td,
                 dockerfile=ctx.default_container.dockerfile,
             )
-            protos = _recursive_list(td)
-            if ctx.workflow_type == WorkflowType.latchbiosdk and remote:
-                local_td = stack.enter_context(tempfile.TemporaryDirectory())
-                scp = SCPClient(
-                    transport=ctx.ssh_client.get_transport(), sanitize=lambda x: x
-                )
-                scp.get(f"{td}/*", local_path=local_td, recursive=True)
-                protos = _recursive_list(local_td)
+
+            if remote:
+                local_td = Path(stack.enter_context(tempfile.TemporaryDirectory()))
+
+                assert scp is not None
+                scp.get(f"{td}/*", local_path=str(local_td), recursive=True)
             else:
-                protos = _recursive_list(td)
+                local_td = Path(td)
+
+            protos = _recursive_list(local_td)
 
             for task_name, container in ctx.container_map.items():
                 task_td = stack.enter_context(MaybeRemoteDir(ctx.ssh_client))
                 try:
-                    build_and_serialize(
+                    _build_and_serialize(
```

Comment:
I got rid of the underscore for hidden function names because we were using them inconsistent and it makes things harder to read